src/components/date/date.component.js

```diff
@@ -24,350 +31,384 @@ import Textbox from "../textbox";
-  const inputRef = useRef();
-  const alreadyFocused = useRef(false);
-  const isBlurBlocked = useRef(false);
-  const focusedViaPicker = useRef(false);
-  const l = useLocale();
-  const { dateFnsLocale } = l.date;
-  const { format, formats } = useMemo(() => getFormatData(dateFnsLocale()), [
-    dateFnsLocale,
-  ]);
-  const { inputRefMap, setInputRefMap } = useContext(DateRangeContext);
-  const inputName = dataElement?.split("-")[0];
-  const [open, setOpen] = useState(false);
-  const [selectedDays, setSelectedDays] = useState(
-    checkISOFormatAndLength(value)
-      ? parseISODate(value)
-      : parseDate(format, value)
-  );
-  const isInitialValue = useRef(true);
-
-  const computeInvalidRawValue = (inputValue) =>
-    allowEmptyValue && !inputValue.length ? inputValue : null;
-
-  const buildCustomEvent = (ev) => {
-    const { id, name } = ev.target;
-
-    const [matchedFormat, matchedValue] = findMatchedFormatAndValue(
-      ev.target.value,
-      formats
-    );
-
-    const formattedValueString =
-      ev.type === "blur"
-        ? formattedValue(format, selectedDays)
-        : ev.target.value;
-    const rawValue = isDateValid(parseDate(matchedFormat, matchedValue))
-      ? formatToISO(...additionalYears(matchedFormat, matchedValue))
-      : computeInvalidRawValue(ev.target.value);
-
-    ev.target = {
-      ...(name && { name }),
-      ...(id && { id }),
-      value: {
-        formattedValue: formattedValueString,
-        rawValue,
-      },
-    };
-
-    return ev;
-  };
-
-  const handleClickAway = () => {
-    if (open) {
-      alreadyFocused.current = true;
-      inputRef.current.focus();
-      isBlurBlocked.current = false;
-      inputRef.current.blur();
-      setOpen(false);
-      alreadyFocused.current = false;
-    }
-  };
-
-  const handleClickInside = useClickAwayListener(handleClickAway, "mousedown");
-
-  const handleChange = (ev) => {
-    isInitialValue.current = false;
-    onChange(buildCustomEvent(ev));
-  };
-
-  const focusInput = () => {
-    focusedViaPicker.current = true;
-    inputRef.current?.focus();
-  };
-
-  const handleDayClick = (day, ev) => {
-    setSelectedDays(day);
-    onChange(
-      buildCustomEvent({
-        ...ev,
-        target: {
-          ...ev.target,
-          value: formattedValue(format, day),
-        },
-      })
+let deprecateInputRefWarnTriggered = false;
+
+const DateInput = React.forwardRef(
+  (
+    {
+      adaptiveLabelBreakpoint,
+      allowEmptyValue,
+      autoFocus,
+      "data-component": dataComponent,
+      "data-element": dataElement,
+      "data-role": dataRole,
+      disabled,
+      disablePortal = false,
+      helpAriaLabel,
+      labelInline,
+      minDate,
+      maxDate,
+      onBlur,
+      onChange,
+      onClick,
+      onFocus,
+      onKeyDown,
+      pickerProps = {},
+      readOnly,
+      size = "medium",
+      tooltipPosition,
+      value,
+      inputRef,
+      ...rest
+    },
+    ref
```

Comment:
```suggestion
    ref: React.ForwardedRef<HTMLInputElement>;
```

Reply:
this is still a `.js` file so I can't add a type annotation 😄

Reply:
Lol, yeah I've missed that :)